grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ReactorGrpcClientStubFactory.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.client.grpc;
+
+import static com.linecorp.armeria.internal.client.grpc.GrpcClientFactoryUtil.newClientStubCreationException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import com.linecorp.armeria.client.grpc.GrpcClientStubFactory;
+
+import io.grpc.Channel;
+import io.grpc.ServiceDescriptor;
+
+/**
+ * A gRPC client stub factory for <a href="https://github.com/salesforce/reactive-grpc">reactive-grpc</a>.
+ */
+public final class ReactorGrpcClientStubFactory implements GrpcClientStubFactory {
+
+    @Override
+    public ServiceDescriptor findServiceDescriptor(Class<?> clientType) {
+        final String clientTypeName = clientType.getName();
+        if (!clientTypeName.endsWith("Stub")) {
```

Comment:
question: should we also check `CoroutineStub` as we do in `JavaGrpcClientStubFactory`?

Reply:
Because a coroutine stub of Kotlin returns a proper `ServiceDescriptor` by the `JavaGrpcClientStubFactory.findServiceDescriptor()`.
However, `null` is returned by `ReactorGrpcClientStubFactory.findServiceDescriptor()` when a `CoroutineStub` is passed. 
Additionally, `ReactorGrpcClientStubFactory` has a lower priority than `KotlinGrpClientStubFactory` so it does not normally happen that a `CoroutineStub` is passed to this factory.

Reply:
It seems confusing to rely on ordering to resolve these correctly, and fragile with respect to refactoring (what if someone felt like changing to use auto-service for example). It could be nice to make these more self-contained, including checking the base class of the stubs (it was left out when only integrating with grpc-java, but now it could be easy for a dependency to mess up these checks.

Reply:
By rule of thumb, the priority was given to what is used a lot. However, even if the orders are changed, the factories are will not be broken.

>  It could be nice to make these more self-contained, including checking the base class of the stubs 

There is no common pattern here. I think it is valid only for gRPC-Kotlin. Because:
- gRPC-Java, Reactive-gRPC and ScalaPB stub extend `io.grpc.stub.AbstractStub`.
- gRPC-Kotlin extends `AbstractCoroutineStub`